pkg/cmd/infra/deployer/deployer.go

```diff
@@ -38,7 +47,20 @@ func NewCommandDeployer(name string) *cobra.Command {
 		Short: "Run the OpenShift deployer",
 		Long:  longCommandDesc,
 		Run: func(c *cobra.Command, args []string) {
-			if err := deploy(cfg); err != nil {
+			kClient, _, err := cfg.Config.Clients()
+			if err != nil {
+				glog.Fatal(err)
+			}
+
+			if len(cfg.DeploymentName) == 0 {
+				glog.Fatal("deployment is required")
+			}
+
+			if len(cfg.Namespace) == 0 {
```

Comment:
@deads2k can you help dan bring this into line?  

Dan, you may want to split this into two pulls so you can get your api changes in first.